service-base/src/main/java/org/eclipse/hono/service/AbstractProtocolAdapterBase.java

```diff
@@ -416,6 +423,73 @@ protected final void closeClients(final Handler<AsyncResult<Void>> closeHandler)
         return result;
     }
 
+    /**
+     * Adds message properties based on a device's registration information.
+     * <p>
+     * Sets the following properties on the message:
+     * <ul>
+     * <li>Adds the registration assertion found in the
+     * {@link RegistrationConstants#FIELD_ASSERTION} property of the given registration information.</li>
+     * <li>Augments the message with missing (application) properties corresponding to the
+     * {@link RegistrationConstants#FIELD_DEFAULTS} contained in the registration information.</li>
+     * <li>Adds JMS vendor properties if configuration property <em>jmsVendorPropertiesEnabled</em> is set
+     * to {@code true}.</li>
+     * </ul>
+     * 
+     * @param message The message to set the properties on.
+     * @param registrationInfo The values to set.
+     */
+    protected final void addProperties(final Message message, final JsonObject registrationInfo) {
+
+        MessageHelper.addRegistrationAssertion(message, registrationInfo.getString(RegistrationConstants.FIELD_ASSERTION));
+        if (getConfig().isDefaultsEnabled()) {
+            final JsonObject defaults = registrationInfo.getJsonObject(RegistrationConstants.FIELD_DEFAULTS);
+            if (defaults != null) {
+                addDefaults(message, defaults);
+            }
+        }
+        if (getConfig().isJmsVendorPropsEnabled()) {
+            MessageHelper.addJmsVendorProperties(message);
+        }
+    }
+
+    private void addDefaults(final Message message, final JsonObject defaults) {
```

Comment:
missing summary for this method.

Reply:
it is a private method
so far, we haven't been adding JavaDocs to private methods IIRC.